test/Parse/statement_typo.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: %target-typecheck-verify-swift
+
+func statementMisspelledAtTheBegginingOfTheLine() {
+  let test = 5
+  gaurd test > 5 else { return } // expected-error {{did you misspell 'guard'}} {{3-8=guard}} expected-error {{use of unresolved identifier 'gaurd'}}
+  i test > 5 { return } // expected-error {{did you misspell 'if'}} {{3-4=if}} expected-error {{use of unresolved identifier 'i'}}
+  whle test > 5 { return } // expected-error {{did you misspell 'while'}} {{3-7=while}} expected-error {{use of unresolved identifier 'whle'}}
```

Comment:
Can we recover better here?  (Would it make sense to use the nearest-neighbor match to start parsing the corresponding statement kind?)

Reply:
Yes. Fixed. Please check my comment below.